src/Symfony/Component/Workflow/Utils/PlaceEnumerationUtils.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Utils;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class PlaceEnumerationUtils
+{
+    public static function getPlaceKey(string|\UnitEnum $place): string
+    {
+        if ($place instanceof \UnitEnum) {
+            return \get_class($place).'::'.$place->name;
+        }
+
+        return $place;
+    }
+
+    public static function getTypedValue(string $place): string|\UnitEnum
+    {
+        try {
+            return \constant($place);
```

Comment:
What about strings that may look like something that is a constant name in something else than an enum ? String place names can be anything.

Reply:
You're right, I'll add a check to be sure that if `\constant` doesn't throw an exception, we're dealing with an enumeration. :+1: